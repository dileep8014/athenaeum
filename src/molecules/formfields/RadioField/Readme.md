RadioField Example:

```example
  <div>
    <RadioField
      label='Fire extinguisher'
      input={{
        name: 'radio_test',
        value: 'Fire ex'
      }}
    />

    <RadioField
      label='Flame Thrower'
      input={{
        name: 'radio_test',
        value: 'Flamer'
      }}
    />
  </div>
```
