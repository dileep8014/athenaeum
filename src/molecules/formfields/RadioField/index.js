import React from 'react';
import PropTypes from 'prop-types';
import { camelCase } from 'lodash';
import classnames from 'classnames';

import styles from './radio_field.module.scss';

function RadioField(props) {
  const {
    label,
    input,
    radioValue
  } = props;

  const labelClasses = [
    styles['label'],
    input.value === radioValue && styles['checked'],
    input.value && input.value !== radioValue && styles['not-selected'],
  ];

  return (
    <label
      className={classnames(styles['radio-field'])}
      htmlFor={`radio-${camelCase(input.value)}`}
    >
      <input
        type='radio'
        name={input.name}
        id={`radio-${camelCase(input.value)}`}
        className={classnames(styles['radio'])}
        {...input}
      />
      <span className={classnames(...labelClasses)}>{ label }</span>
    </label>
  );
}


RadioField.propTypes = {
  label: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object
  ]),
  radioValue: PropTypes.string.isRequired,
  /**
   * The props under the meta key are metadata about the state of this field that `redux-form` tracks.
   */
  meta: PropTypes.object,
  /**
   * The props under the input key are passed from `redux-form` and spread into `<input />`.
   */
  input: PropTypes.object.isRequired
};

export default RadioField;
