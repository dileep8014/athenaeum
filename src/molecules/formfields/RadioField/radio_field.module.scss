$size: ru(.75);
$gap: ru(.5);
$checked-size: ru(.5);

.radio-field {
  @include typography-6;

  margin: 0;
  display: block;
  position: relative;
  min-height: $size;
  padding-left: $size + $gap;
  user-select: none;

  + .radio-field {
    margin-top: ru(.75);
  }
}

// pseudo radio button
.radio-field:before {
  position: absolute;
  top: 0;
  left: 0;
  width: $size;
  height: $size;
  overflow: hidden;
  margin-top: (ru(1) - $size) / 2;

  border: 1px solid color('neutral-5');
  background-color: color('neutral-8');
  border-radius: 50%;
  content: '';
}

.label {
  display: inline-block;
  color: color('neutral-3');
  cursor: pointer;

  &:after {
    content: '';
    opacity: 0;
    position: absolute;
    top: 0;
    left: 0;
    width: $checked-size;
    height: $checked-size;
    margin-top: (ru(1) - $size) / 2 + ($size - $checked-size) / 2;
    margin-left: ($size - $checked-size) / 2;
    border-radius: 50%;
    background-color: color('brand-3');
  }
}

.radio {
  position: absolute;
  top: 0;
  left: 0;
  width: $size + $gap;
  height: $size;
  opacity: 0;
  margin: 0;
  cursor: pointer;
}

.checked {
    color: color('neutral-2');

    &:after { opacity: 1; }
}

.radio:disabled {
  + .label {
    cursor: not-allowed;
    color: color('neutral-3');

    &:after {
      box-shadow: 2px 2px 0 0 color('neutral-3');
    }
  }
}


@media #{$medium-up} {
  @include typography-7;
  line-height: ru(.75);
  margin: 0;

  .label {
    margin-top: ru(.125);
  }
}
