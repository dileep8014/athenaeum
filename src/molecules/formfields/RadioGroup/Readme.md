### RadioGroup Example:

```jsx
    class RadioGroupExample extends React.Component {
      constructor(props) {
        super(props);

        this.state = { value: '' };

        this.onChange = this.onChange.bind(this);
      }

      onChange(value) {
        this.setState({ value });
      }

      render() {
        const { value } = this.state;

        return (
          <RadioGroup
            label='Test label'
            subLabel='Sublabel action'
            meta={{}}
            input={{}}
            tooltip='Hello friend'
          >
            <RadioField
              label='Burgers'
              radioValue='burgers'
              input={{ name: 'food', value, onChange: this.onChange }}
            />
            <RadioField
              label='Hot Dogs'
              radioValue='dogs'
              input={{ name: 'food', value, onChange: this.onChange }}
            />
          </RadioGroup>
        )
      }
    }

    <RadioGroupExample />
```

### RadioGroup Example (no label):

```jsx
    class RadioGroupExampleNoLabel extends React.Component {
      constructor(props) {
        super(props);

        this.state = { value: '' };

        this.onChange = this.onChange.bind(this);
      }

      onChange(value) {
        this.setState({ value });
      }

      render() {
        const { value } = this.state;

        return (
          <RadioGroup meta={{}} input={{}}>
            <RadioField
              label='Burgers'
              radioValue='burgers'
              input={{ name: 'food', value, onChange: this.onChange }}
            />
            <RadioField
              label='Hot Dogs'
              radioValue='dogs'
              input={{ name: 'food', value, onChange: this.onChange }}
            />
          </RadioGroup>
        )
      }
    }

    <RadioGroupExampleNoLabel />
```
