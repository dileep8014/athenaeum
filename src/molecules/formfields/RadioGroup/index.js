import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import ErrorMessage from 'atoms/ErrorMessage';
import Text from 'atoms/Text';
import { renderTooltip } from 'utils/fieldUtils';

import styles from './radio_group.module.scss';

class RadioGroup extends React.Component {
  render() {
    const {
      children,
      className,
      label,
      subLabel,
      input,
      meta,
      tooltip,
      fieldRef,
    } = this.props;

    const showErrorMessage = !meta.active && (meta.visited || meta.submitFailed);
    const message = meta && showErrorMessage && (meta.error || meta.warning);

    const classes = [
      styles['radio-group'],
      meta && meta.active && styles['focused'],
      meta && showErrorMessage && meta.error && !meta.active && styles['hasError'],
      className,
    ];

    return (
      <div ref={fieldRef && fieldRef}>
        <div
          className={classnames(...classes)}
          onBlur={(e) => {
            if (!e.relatedTarget) input.onBlur();
          }}
          onFocus={input && input.onFocus}
        >
          <div className={styles['label-wrapper']}>
            <div className={styles['label']}>
              { /* eslint-disable jsx-a11y/label-has-for */ }
              <label htmlFor={input.name}>{label}</label>
              { /* eslint-enable jsx-a11y/label-has-for */ }
              { tooltip && renderTooltip(tooltip, styles['tooltip'], styles['tooltip-icon']) }
            </div>

            {
              subLabel && (
                <Text
                  size={10}
                  font='b'
                >
                  {subLabel}
                </Text>
              )
            }
          </div>
          {
            React.Children.map(children, (child) => (
              <div className={styles['field']}>
                { child }
              </div>
            ))
          }
        </div>

        <ErrorMessage
          condition={!!message}
          message={message}
        />
      </div>
    );
  }
}

RadioGroup.propTypes = {

  /**
   * This prop will add a new className to any inherent classNames
   * provided in the component's index.js file.
   */
  className: PropTypes.string,

  /**
   * Label for group
   */
  label: PropTypes.string,

  /**
   * label to appear directly under the label
   */
  subLabel: PropTypes.string,

  /**
   * Input props provided by ReduxForm
   */
  input: PropTypes.object,

  /**
   * meta props provided by ReduxForm
   */
  meta: PropTypes.object,

  /**
   * either a handler for clicking the tooltip, or text to go in the tooltip for the label
   */
  tooltip: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.string,
  ]),

  /**
   * Applies a React ref to the wrapping node for this field
   */
  fieldRef: PropTypes.func,
};

export default RadioGroup;
