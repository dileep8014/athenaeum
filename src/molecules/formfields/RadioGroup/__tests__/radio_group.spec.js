/* eslint-disable no-extra-parens */
import React from 'react';
import renderer from 'react-test-renderer';

import RadioGroup from 'molecules/formfields/RadioGroup';
import RadioField from 'molecules/formfields/RadioField';

describe('<RadioGroup />', () => {
  let component;

  beforeEach(() => {
    component = (
      <RadioGroup
        label='Test label'
        subLabel='Sublabel action'
        meta={{}}
        input={{
          value: 'test',
        }}
        tooltip='Hello friend'
      >
        <RadioField
          label='Burgers'
          radioValue='burgers'
          input={{
            name: 'food',
            value: 'dogs',
            onChange: jest.fn(),
          }}
        />
        <RadioField
          label='Hot Dogs'
          radioValue='dogs'
          input={{
            name: 'food',
            value: 'dogs',
            onChange: jest.fn(),
          }}
        />
      </RadioGroup>
    );
  });

  it('renders correctly', () => {
    const actual = renderer.create(component).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
