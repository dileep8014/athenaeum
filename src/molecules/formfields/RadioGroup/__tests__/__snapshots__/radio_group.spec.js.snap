// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup /> renders correctly 1`] = `
<div>
  <div
    className="radio-group"
    onBlur={[Function]}
    onFocus={undefined}
  >
    <div
      className="header"
    >
      <div
        className="label-wrapper"
      >
        <label
          className="label"
          htmlFor={undefined}
        >
          Test label
        </label>
        <span>
          <span
            className="tooltip-wrapper tooltip"
            onClick={[Function]}
          >
            <span
              className="icon-wrapper tooltip-icon"
              onClick={undefined}
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            >
              <img
                alt="tooltip"
                className="img"
                data-src={undefined}
                src="https://static.policygenius.com/svgs/tooltip.svgz"
                title="tooltip"
              />
            </span>
            <span
              className="hover-message right"
            >
              Hello friend
            </span>
          </span>
        </span>
      </div>
      <p
        className="primary-3 type-b-10-medium inherit"
      >
        Sublabel action
      </p>
    </div>
    <div
      className="field"
    >
      <label
        className="radio-field"
        htmlFor="radio-burgers"
      >
        <input
          className="radio"
          id="radio-burgers"
          name="food"
          onChange={[Function]}
          type="radio"
          value="dogs"
        />
        <span
          className="label not-selected"
        >
          Burgers
        </span>
      </label>
    </div>
    <div
      className="field"
    >
      <label
        className="radio-field"
        htmlFor="radio-dogs"
      >
        <input
          className="radio"
          id="radio-dogs"
          name="food"
          onChange={[Function]}
          type="radio"
          value="dogs"
        />
        <span
          className="label checked"
        >
          Hot Dogs
        </span>
      </label>
    </div>
  </div>
  <span />
</div>
`;
